pkg/apis/hive/v1alpha1/validating-webhooks/clusterdeployment_validating_admission_hook.go

```diff
@@ -590,6 +602,56 @@ func validateDomain(domain string, validDomains []string) bool {
 	return false
 }
 
+func validateMachinePools(newObject *hivev1.ClusterDeployment, contextLogger *log.Entry) *admissionv1beta1.AdmissionResponse {
```

Comment:
The logger is never used. Do we need to pass it to the function?

Reply:
done